packages/expo/src/ScreenOrientation/ScreenOrientation.ts

```diff
@@ -76,20 +51,25 @@ export async function lockAsync(orientationLock: OrientationLock): Promise<void>
     throw new TypeError(`Invalid Orientation Lock: ${orientationLock}`);
   }
 
-  if (orientationLock === OrientationLock.OTHER){
+  if (orientationLock === OrientationLock.OTHER) {
     return;
   }
 
   await ExpoScreenOrientation.lockAsync(orientationLock);
+  _lastOrientationLock = orientationLock;
 }
 
 export async function lockPlatformAsync(options: PlatformOrientationInfo): Promise<void> {
   if (!ExpoScreenOrientation.lockPlatformAsync) {
     throw new UnavailabilityError('ScreenOrientation', 'lockPlatformAsync');
   }
 
-  const { screenOrientationConstantAndroid, screenOrientationArrayIOS } = options;
-  let platformOrientationParam;
+  const {
+    screenOrientationConstantAndroid,
+    screenOrientationArrayIOS,
+    screenOrientationArrayWeb,
```

Comment:
Instead of an array, this would be `screenOrientationLockWeb` and can be any value from the modern web API https://w3c.github.io/screen-orientation/#dom-orientationlocktype